Спринт №17

Keithcutt opened the pull request:
Внедрение зависимостей с помощью Koin.

Keithcutt committed:
Домашнее задание №16

Инициализация всех ключевых объектов и внедрение их как зависимостей переведены на Koin.

Исправлен цвет фона экрана плеера на корректный.

ArturNurtdinov commented on app/src/main/java/com/example/playlistmaker/player/presentation/mapper/TrackMapper.kt:
```diff
@@ -0,0 +1,17 @@
+
+    fun getFromIntent(intent: Intent): Track {
+        val json = intent.getStringExtra(TRACK_KEY)
+        return Gson().fromJson(json, Track::class.java)
```
У тебя `Gson` уже есть в графе зависимости, в этот класс его тоже нужно заинжектить

ArturNurtdinov commented on app/src/main/java/com/example/playlistmaker/settings/di/SettingsModule.kt:
```diff
@@ -0,0 +1,28 @@
+    }
+
+    single<SettingsRepository> { SettingsRepositoryImpl(get()) }
+    single<SettingsInteractor> { SettingsInteractorImpl(get()) }
```
Интеракторы в общем случае не являются тяжёлыми объектами и не хранят состояние, их очень легко создать. Поэтому их не принято делать синглами, а при необходимости создавать новый объект, чтобы не тратить лишнюю память на их хранение, поэтому здесь лучше сделать через `factory`.

Keithcutt committed:
Работа над ошибками:
Gson для классов TrackMapper и SearchHistoryRepositoryImpl теперь создается в Koin и передается в эти классы в качестве зависимостей.

Интеракторы в модулях создаются через factory.
